pkg/dockercreds/dockercreds_test.go

```diff
@@ -121,9 +122,11 @@ type ReadKubernetesDockerRegistrySecretTestSuite struct {
 
 func (suite *ReadKubernetesDockerRegistrySecretTestSuite) SetupTest() {
 	suite.DockerCredsTestSuite.SetupTest()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
Make sure this doesn't affect any other tests, as this will cancel the goroutine when setup finishes